internal/token/token.go

```diff
@@ -0,0 +1,53 @@
+// Package token provides functions to save and load tokens from disk.
+package token
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/ubuntu/authd-oidc-brokers/internal/providers/info"
+	"golang.org/x/oauth2"
+)
+
+// AuthCachedInfo represents the token that will be saved on disk for offline authentication.
+type AuthCachedInfo struct {
+	Token      *oauth2.Token
+	RawIDToken string
+	UserInfo   info.User
+}
+
+// CacheAuthInfo saves the token to the given path.
+func CacheAuthInfo(path string, token AuthCachedInfo) (err error) {
```

Comment:
Maybe a better name for the `AuthCachedInfo` argument? `token` could be misleading since we store more things than just the token itself.

Reply:
I actually wanted to propose to rename `CacheAuthInfo` / `LoadAuthInfo` to `StoreToken` / `LoadToken`, and to refer to the content of the `token.json` file as "token" instead of "AuthInfo" or "CachedInfo" (as we already do in many comments and log messages). Yes, that content is more than just the *OAuth2* token, but I think we can still refer to it as our (authd broker) token. What do you think? Would also be interested in @didrocks opinion.

Reply:
Sounds good. Also relates to my other comment about splitting the struct into two different ones.

Reply:
I would like to hear @didrocks opinion before I implement this, because it will change a lot of lines.  We might also want to do that in a separate PR, which we can review and merge quickly to avoid merge conflicts.

Reply:
I think indeed, now that token grew into its own separate entity, this is really what we got from msentraid, and we a more utility base approach like Store/Load sounds good to me.